apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -63,37 +220,69 @@ init(#{parent := Parent, chan_id := ChanId, requests := Reqs}) ->
     self() ! check_status,
     {ok, #st{parent = Parent, chan_id = ChanId, requests = Reqs}}.
 
-handle_info({gproc_ps_event, top_changed, _}, #st{} = St) ->
-    {noreply, check_status(St)};
-handle_info({gproc_ps_event, {tx_event, E}, I}, St) ->
-    lager:debug("tx_event (ignored): ~p, I = ~p", [E, I]),
-    {noreply, St};
+
+%% Strategy for monitoring the chain.
+%%
+%% Challenges:
+%% 1. There is currently no revenue model for the SC FSM, i.e. the node
+%%    running the FSM has no way of getting reimbursed for the cost.
+%%    This means the FSM *and* the watcher must be as lightweight as possible.
+%% 2. The watcher must keep up with fork switches.
+%%
+%% To allow for lightweight monitoring, we use custom tx events keyed on the
+%% channel id. These are generated at block insertion, and are published
+%% *before* any corresponding `top_changed' event.
+%%
+%% We keep the tx events in a sliding window LIFO buffer. For each `top_changed'
+%% event, we need to consider the following scenarios:
+%% 1. The new block contains a tx for our channel: Read the channel object
+%%    and report accordingly
+%% 2. We are waiting for a minimum-depth condition: Verify that it's still
+%%    relevant, and check the depth.
+%% 3. A fork switch may have happened, changing the channel object even if
+%%    no channel tx is present in the top block: Dig into the chain to figure
+%%    out the current channel state. Remove txes from the tx_log that have
+%%    been returned to the mempool.
+%% 4. No fork switch, no channel tx in the top block, and no depth watch:
+%%    Do nothing.
+handle_info({gproc_ps_event, top_changed, #{info := Info}}, #st{} = St) ->
+    lager:debug("(Fsm = ~p) top_changed: ~p", [St#st.parent, Info]),
+    {noreply, check_status(Info, St)};
+handle_info({gproc_ps_event, {tx_event, {channel, ChId}},
+             #{info := #{ block_hash := _BlockHash
+                        , tx_hash    := _TxHash
+                        , type       := _Type } = I}}, #st{chan_id = ChId} = St) ->
+    lager:debug("tx_event: I = ~p", [I]),
+    {noreply, log_tx(I, St)};
 handle_info({'DOWN', _, process, Parent, _}, #st{parent = Parent} = St) ->
     {stop, normal, St};
 handle_info(check_status, St) ->
-    {noreply, check_status(St)};
+    {noreply, check_status(top, St)};
```

Comment:
I see that on reception of `check_status` message the function `check_status/2` is called with `top` as first argument (later called "scenario"). Ok.